int32 obj_idx
---
bool ack